### PR TITLE
Setup Storybook and deploy to GH pages

### LINKED ISSUE #1: Setup Storybook and deploy it using GH Pages
An example of working storybook: https://github.com/datopian/datapub
Deployed to gh pages: https://datopian.github.io/datapub/

## Acceptance criteria

* [x] I have storybook running here https://datopian.github.io/data-explorer-graphql/ __Done__  https://github.com/datopian/data-explorer-graphql/pull/7

## Tasks

* [x] Setup Storybook with some fixtures for this project
* [x] Deploy to gh pages

### PR DESCRIPTION
**Pull request by risenW**
This pull fixes #1 

You can preview the current storybook [here](https://datopian.github.io/data-explorer-graphql/?path=/story/templates--app-template)

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -9,7 +9,7 @@
     <noscript>You need to enable JavaScript to run this app.</noscript>
     <div
       id="root"
-      data-graphql="http://localhost:3333/v1/graphql"
+      data-graphql="http://localhost:3000/v1/graphql"
```
Comment (anuveyatsu) [inline public/index.html:12]: @risenW I believe you have React app running on port 3000 so I wouldn't expect to have GraphQL endpoint at the port.

Comment (anuveyatsu) [reply public/index.html:12]: But it's ok to leave it as this.

File: package.json [hunk 2]
```diff
@@ -9,17 +9,20 @@
     "@testing-library/user-event": "^12.1.10",
     "graphql": "^15.4.0",
     "graphql-query-builder": "^1.0.7",
-    "react": "^17.0.1",
-    "react-dom": "^17.0.1",
-    "react-scripts": "4.0.1",
+    "react": "^16.13.0",
```
Comment (anuveyatsu) [inline package.json:12]: @risenW why are we downgrading here? Is it Storybook requirement?

File: package.json [hunk 3]
```diff
@@ -9,17 +9,20 @@
...
+    "react-dom": "^16.13.0",
```
Comment (anuveyatsu) [inline package.json:13]: Same here

File: package.json [hunk 4]
```diff
@@ -9,17 +9,20 @@
...
+    "react-scripts": "3.4.3",
```
Comment (anuveyatsu) [inline package.json:14]: Same here

File: .github/workflows/node.js.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Set up Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Install dependencies
+      run: npm install
```
Comment (anuveyatsu) [inline .github/workflows/node.js.yml:31]: We're using `yarn` by default. Also, `npm install` wouldn't check our `yarn.lock` file.

File: .github/workflows/node.js.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+    - name: Run the tests
+      run: npm test
```
Comment (anuveyatsu) [inline .github/workflows/node.js.yml:34]: Let's use `yarn` here as well.

File: .github/workflows/node.js.yml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+    - name: Build
+      run: CI=false npm run build
```
Comment (anuveyatsu) [inline .github/workflows/node.js.yml:37]: Same here re `yarn`

File: .github/workflows/node.js.yml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+    - name: Deploy storybook to Github Pages
+      run: npm run deploy-storybook -- --ci
```
Comment (anuveyatsu) [inline .github/workflows/node.js.yml:40]: And here.

**Review by anuveyatsu — CHANGES_REQUESTED** [review]
Left some comments @risenW

File: public/index.html (hunk 1 above)
Comment (risenW) [reply public/index.html:12]: I have react running in 3001 locally though

File: package.json (hunk 2 above)
Comment (risenW) [reply package.json:12]: Yea, it's a storybook requirement. Ref https://www.learnstorybook.com/intro-to-storybook/react/en/get-started/

File: package.json (hunk 4 above)
Comment (risenW) [reply package.json:14]: Storybook requirement as well

File: .github/workflows/node.js.yml (hunk 5 above)
Comment (risenW) [reply .github/workflows/node.js.yml:31]: Thanks, fixing this now

**Review by anuveyatsu — APPROVED** [review]

**Pull request merged**